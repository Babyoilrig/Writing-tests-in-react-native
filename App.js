
import { useState } from "react";
import { Image, StyleSheet, Text, View, Button } from "react-native";
import catandmartha from './assets/cat_and-martha.jpg';




export default function App() {
  //State to change text
  const [outputText, setOutputText] = useState('Hello Martha!');

  
  return (
    <View style={styles.container}>
    <Image source={catandmartha} testID= "catAndMarthaImage" style={{ width: 305, height: 305 }} />
      <Text style={{color:'#FEDD00', fontSize: 36}}>{outputText}</Text>
      <Button
        title="Change text"
        onPress={() => setOutputText("You clicked me!")}
      />
      <Text>"Hello World!</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#FB48C4",
    alignItems: "center",
    justifyContent: "center",
  },
});

