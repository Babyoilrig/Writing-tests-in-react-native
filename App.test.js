import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

import App from './App';


describe("App created", () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

    it('should display Hello World', () => {
      const text = tree.root.findByProps({testID: "myButton"}).props;
      expect(text.children).toEqual('Hello World');
    });
  
});

//Was up to: Watching youtube vid - 4.48 ish 
//Says tree is not defined - in vid it is imported
//Trying to figure out how to define tree in this file - 
//so that we can run the test - cos atm it's failing