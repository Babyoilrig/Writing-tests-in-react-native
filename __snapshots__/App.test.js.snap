// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App created renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="myText"
  >
    Hello World!
  </Text>
</View>
`;

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World!
  </Text>
</View>
`;

exports[`App should render Hello World! renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World!
  </Text>
</View>
`;
